core/drivers/plic.c

```diff
@@ -0,0 +1,277 @@
+// SPDX-License-Identifier: BSD-2-Clause
+/*
+ * Copyright 2022-2023 NXP
+ *
+ */
+
+#include <assert.h>
+#include <drivers/plic.h>
+#include <config.h>
+#include <io.h>
+#include <kernel/dt.h>
+#include <kernel/interrupt.h>
+#include <kernel/panic.h>
+#include <mm/core_memprot.h>
+#include <mm/core_mmu.h>
+#include <platform_config.h>
+#include <trace.h>
+
+#ifdef CFG_RISCV_PLIC
+
+#define PLIC_PRIORITY_OFFSET		0
+#define PLIC_PENDING_OFFSET		0x1000
+#define PLIC_ENABLE_OFFSET		0x2000
+#define PLIC_THRESHOLD_OFFSET		0x200000
+#define PLIC_CLAIM_OFFSET		0x200004
+
+#define PLIC_PRIORITY_SHIFT_PER_SOURCE	U(2)
+#define PLIC_PENDING_SHIFT_PER_SOURCE	U(0)
+
+#define PLIC_ENABLE_SHIFT_PER_TARGET	U(7)
+#define PLIC_THRESHOLD_SHIFT_PER_TARGET	U(12)
+#define PLIC_CLAIM_SHIFT_PER_TARGET	U(12)
+
+#define PLIC_PRIORITY(base, source) \
+	((base) + PLIC_PRIORITY_OFFSET +\
+	SHIFT_U32(source, PLIC_PRIORITY_SHIFT_PER_SOURCE))
```

Comment:
indentation  (1 space char before `SHIFT_U32`) + parantheses around `source`:
```c
#define PLIC_PRIORITY(base, source) \
	((base) + PLIC_PRIORITY_OFFSET + \
	 SHIFT_U32((source), PLIC_PRIORITY_SHIFT_PER_SOURCE))
```
Indentation simpler with:
```c
#define PLIC_PRIORITY(base, source) ( \
		(base) + PLIC_PRIORITY_OFFSET + \
	 	SHIFT_U32((source), PLIC_PRIORITY_SHIFT_PER_SOURCE) \
	)
```


Same kind of comments for the below macro definitions.